src/vs/workbench/parts/files/browser/editors/textFileEditor.ts

```diff
@@ -67,6 +70,12 @@ export class TextFileEditor extends BaseTextEditor {
 		}
 	}
 
+	private onWillCloseEditor(e: IEditorIdentifier): void {
+		if (e.editor === this.input && this.position === this.editorGroupService.getStacksModel().positionOfGroup(e.group)) {
```

Comment:
@isidorn since last milestone, file editor inputs are actually shared when they have the same URI, you can see that here: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/services/editor/browser/editorService.ts#L264